actionpack/lib/action_dispatch/system_test_case.rb

```diff
@@ -89,6 +89,20 @@ module ActionDispatch
   #       { js_errors: true }
   #   end
   #
+  # Most drivers won't let you add specific browser capabilities through the +options+ mentioned above.
```

Comment:
I think we should deprecate the options hash if this really doesn't work most of the time. It's confusing to have 2 ways and one way maybe not work. What do you think?

Reply:
Yea I think the phrasing is confusing for new users though. How do I know when to use the options hash and when to use the block version? Does Selenium clearly document what works and doesnt?

Reply:
Correct, `options` previously contained a lot of various things, and some of them has been deprecated in favor of the new `Options` object with a better and less "throw it all" API (see https://github.com/SeleniumHQ/selenium/blob/29b5be0a72331df9ab99e36b05b4c29cf7046f30/rb/lib/selenium/webdriver/chrome/driver.rb#L67)

Reply:
> Yea I think the phrasing is confusing for new users though. 

Agree it's confusing. The problem is that depending on the driver chose by the user, passing capabilities is done differently. On firefox and chrome `<Driver>::Options` is responsible for it (and thus you can't pass anymore a hash of options to the driver) where as on Microsoft Edge, it's still done directly through the driver.

The `<Driver>::Options` class on selenium is still recent and I think the long term goal would be to have one for each driver.

I haven't found documentation on what "works and what doesn't". However, if an app used to pass capabilities through the `options` hash, then he'll get deprecation warning from selenium and explicitly tell to use Options.
As any deprecations if the app doesn't fix it, it will break as soon as selenium gets updated.